lib/mongo/socket.rb

```diff
@@ -169,11 +169,27 @@ def eof?
     private
 
     def read_from_socket(length)
-      data = String.new
+      data = ('x'*length).force_encoding('BINARY')
```

Comment:
Does the encoding of the final `data` string matter?

Reply:
It shouldn't since the contents should be binary data anyway.

Reply:
I assume the character `x` is just a placeholder here so that we can preallocate the string? I think a comment explaining why we're doing this might be useful.